### PR TITLE
feat(client): allow users to update project and service name in the home page itself

### PR DESCRIPTION
**Pull request by lokeswaran-aj**
Close: #4559 & #4457

## PR Details

I have added a form to update the project or service name without going to the settings page. 

https://user-images.githubusercontent.com/74011196/217592133-6c47ce1b-90a9-4fc7-a7a0-b14c564fbf13.mp4

I tried my level best to achieve what was given in the Figma design in the issue. But I don't have enough knowledge to implement the following
- add custom error message(eg: Unsupported character)
- show error as a pointing label(like in Figma design)

Please comment if you need to see any other changes. Also if you feel the above checks are a must, please provide me with some documentation to fix them. I will try.

## PR Checklist

- [x] Tests for the changes have been added
- [x] `npm test` doesn't throw any error

### CODE REVIEW

File: packages/amplication-client/src/Resource/ResourceHome.tsx [hunk 1]
```diff
@@ -50,7 +114,60 @@ const ResourceHome = ({ match, innerRoutes }: Props) => {
                 resourceThemeMap[currentResource?.resourceType].color,
             }}
           >
-            {currentResource?.name}
+            <div className={`${CLASS_NAME}__header__input__container`}>
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Resource/ResourceHome.tsx:117]: pelase use a seperate component for this form

File: packages/amplication-client/src/Resource/ResourceHome.tsx [hunk 2]
```diff
@@ -50,7 +114,60 @@ const ResourceHome = ({ match, innerRoutes }: Props) => {
...
+                              icon="info_circle"
+                              size="small"
+                              className={`${CLASS_NAME}__invalid`}
+                            />
+                          )}
+                        </Form>
+                      ) : (
+                        <span
+                          className={`${CLASS_NAME}__header__text`}
+                          onClick={() => setIsEditing(true)}
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Resource/ResourceHome.tsx:158]: please use useCallback to define the function

**Review by yuval-hazaz — CHANGES_REQUESTED** [review]
looks good
please create a seperate component so it will be easier to review and maintain
you can call it /Resource/ResourceNameField.tsx

File: packages/amplication-client/src/Resource/ResourceHome.tsx (hunk 1 above)
Comment (lokeswaran-aj) [reply packages/amplication-client/src/Resource/ResourceHome.tsx:117]: Sure, will work on it

File: packages/amplication-client/src/Resource/ResourceHome.tsx (hunk 2 above)
Comment (lokeswaran-aj) [reply packages/amplication-client/src/Resource/ResourceHome.tsx:158]: Can you be specific on using useCallback on which function, please?

**Comment by lokeswaran-aj** [comment]
Hi, @yuval-hazaz Move the code to the new component `ResouceNamefield`. And `handleSubmit` already uses useCallback, so I am not sure where else to utilize useCallback. Can you review the code and update me?

**Comment by lokeswaran-aj** [comment]
@yuval-hazaz Did you find time to review the changes?

**Comment by abrl91** [comment]
run branch

**Review by yuval-hazaz — APPROVED** [review]
well done

**Comment by yuval-hazaz** [comment]
@souravjain540 it would be great if we could send @lokeswaran-aj some swag in case we didn't do it already 💜

**Comment by lokeswaran-aj** [comment]
> well done
Thank you @yuval-hazaz This is the first feature request or enhancement PR that I have created in my life.




> @souravjain540 it would be great if we could send @lokeswaran-aj some swag in case we didn't do it already 💜

@souravjain540  I m looking forward to some awesome swag🚀

BTW, 2 checks are failing for unchanged code. Do I have to make any changes or will you merge it?

**Pull request merged**